### PR TITLE
Update Yii 2 controller matching

### PR DESCRIPTION
**Pull request by morrisonlevi**
### Description

Use the first controller that matches the Application->requestedRoute as the app.endpoint. Before, it was using the first controller whose post-hook executed first.

### Readiness checklist
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [ ] Changelog has been added to the appropriate release draft. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: src/DDTrace/Integrations/Yii/V2/YiiIntegrationLoader.php [hunk 1]
```diff
@@ -37,6 +37,22 @@ function (SpanData $span) use ($service) {
...
+        $first_controller = null;
+        \dd_trace_method(
+            'yii\base\Module',
+            'createController',
+            function (SpanData $span, $args, $retval, $ex) use (&$first_controller) {
+                if (!$ex && isset($args[0], $retval) && \is_array($retval) && !empty($retval)) {
+                    $first_controller = $retval[0];
+                    if ($this->requestedRoute === $args[0]) {
+                        $first_controller = $retval[0];
+                    }
```
Comment (labbati) [inline src/DDTrace/Integrations/Yii/V2/YiiIntegrationLoader.php]: I think these lines have no effect.

Comment (morrisonlevi) [reply src/DDTrace/Integrations/Yii/V2/YiiIntegrationLoader.php]: The `$first_controller` is captured by reference, and is then used in `yii\base\Controller::runAction`.

Comment (labbati) [reply src/DDTrace/Integrations/Yii/V2/YiiIntegrationLoader.php]: What I mean is that line 49 has no effects as line 47 (identical) has already been executed

Comment (morrisonlevi) [reply src/DDTrace/Integrations/Yii/V2/YiiIntegrationLoader.php]: Ah, I see. It's line 47 that should be removed. Oops!

Comment (labbati) [reply src/DDTrace/Integrations/Yii/V2/YiiIntegrationLoader.php]: 👍 now it makes sense 😄

**Pull request merged**